### PR TITLE
fix(Scripts/Gruuls Lair): Improvements to High King Maulgar

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Change cast time to 7s
-  

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/4813
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14720
-  closes https://github.com/azerothcore/azerothcore-wotlk/issues/14718
* closes https://github.com/azerothcore/azerothcore-wotlk/issues/14719

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

**Comment by balleny** [comment]
should this really be
`if (Unit* target = SelectTarget(SelectTargetMethod::MaxThreat, 1))`
and not better be the current threat holder?
(given the max threat person is getting sheeped, second person in threat table is attacked. the next cast (given the sheep can overlap, which I think shouldnt be possible) the 2nd person in threat, who is tanking currently, wouldnt get sheeped - but the already sheeped player would be re-sheeped

**Comment by Kitzunu** [comment]
The only problem I see with it is that using GetVictim() you could taunt swap it to a dps or something neglecting the mechanic.

I think generally someone wouldn't retain threat if you are crowd controlled and the group isnt just standing doing nothing

**Comment by balleny** [comment]
Following the regular boss tactics, this npc is usually not killed first. Typically, you utilize a moonkin (immune to sheep) or two casters dedicated for tanking.
But, often also a felhunter summoned by another npc during the bossfight is used for tanking. That one is typically generating more threat than a healer, but not enough to overtake the original ranged tanking (which might be sheeped already (didnt check the timers))

**Comment by Kitzunu** [comment]
Test it. It is either checking MaxThreat, or GetVictim() which means you can taunt swap before the cast to get the sheep on someone else

i.e tank tanks it, right before cast a warrior taunts it getting the CC, which then the tank can taunt again

File: src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp [hunk 1]
```diff
@@ -382,7 +382,7 @@ class boss_blindeye_the_seer : public CreatureScript
             me->SetInCombatWithZone();
             instance->SetBossState(DATA_MAULGAR, IN_PROGRESS);
 
-            events.ScheduleEvent(EVENT_ADD_ABILITY1, 5000);
+            events.ScheduleEvent(EVENT_ADD_ABILITY1, 1);
```
Comment (Gultask) [inline src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: Uneducated guess, but wouldn't this mean that the ability is cast within 1ms of combat starting?

Comment (balleny) [reply src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: yes, and I have yet to find a video where the PW:S is casted practically on pull

Comment (Kitzunu) [reply src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: I understand the issue as "Spell Shield buff as first cast on pull" means it should be cast on pull?

Comment (balleny) [reply src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: most of the discussion is in the initial ticket https://github.com/chromiecraft/chromiecraft/issues/4811
long story short, the cast should occur within 1-2 seconds of the pull - but be the first spell casted (must happen before the first fireball and blastnova
see also
https://github.com/chromiecraft/chromiecraft/issues/4873

Comment (Gultask) [reply src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: 1.2s if wanting to replicate Blizz queueing and packet cycles

Comment (Gultask) [reply src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: But honestly that whole part is a mess. Tying 4 creatures' spells together. Scripting made by insane people that is.

Comment (Kitzunu) [reply src/server/scripts/Outland/GruulsLair/boss_high_king_maulgar.cpp:385]: I can fix it that way.

They are not tied together tho, just named the same x)

**Comment by Kitzunu** [comment]
Okay, got the cast set to 1.7s

**Comment by Maelthyrr** [comment]
Hi, tested and no issues find.

**Pull request merged**